local-cli/server/util/attachHMRServer.js

```diff
@@ -16,11 +16,254 @@ const url = require('url');
+                  dependencies: modulesToUpdate,
+                });
+              });
+
+          });
+        })
+        .then((resolutionResponses) => {
+          if (clients.length === 0 || !resolutionResponses) {
+            return;
+          }
+
+          return forEachBundleEntry((bundleEntry, platform) => {
+            if (!resolutionResponses[platform] || !resolutionResponses[platform][bundleEntry]) {
+              return;
+            }
+
+            if (!platformCache[platform][bundleEntry].shallowDependencies[filename]){
+              return;
+            }
+
+            const resolutionResponse = resolutionResponses[platform][bundleEntry];
+
+            const httpServerAddress = httpServer.address();
+
+            // Sanitize the value from the HTTP server
+            let packagerHost = 'localhost';
+            if (httpServerAddress.address &&
+                httpServerAddress.address !== '::' &&
+                httpServerAddress.address !== '') {
+              packagerHost = httpServerAddress.address;
+            }
+
+            let packagerPort = httpServerAddress.port;
+            return packagerServer.buildBundleForHMR({
+              entryFile: bundleEntry,
+              platform,
+              resolutionResponse,
+            }, packagerHost, packagerPort);
+          });
+        })
+        .then(bundles => {
+          if (clients.length === 0 || !bundles) {
+            return;
+          }
+
+          return forEachBundleEntry((bundleEntry, platform) => {
+            if (!bundles[platform]) {
+              return;
+            }
+
+            const bundle = bundles[platform][bundleEntry];
+
+            if (!bundle || bundle.isEmpty()) {
+              return;
+            }
+            return {
+              type: 'update',
+              body: {
+                modules: bundle.getModulesCode(),
+                sourceURLs: bundle.getSourceURLs(),
+                sourceMappingURLs: bundle.getSourceMappingURLs(),
+              },
+            };
+          });
+        })
+        .catch(error => {
+          // send errors to the client instead of killing packager server
+          let body;
+          if (error.type === 'TransformError' ||
+              error.type === 'NotFoundError' ||
+              error.type === 'UnableToResolveError') {
+            body = {
+              type: error.type,
+              description: error.description,
+              filename: error.filename,
+              lineNumber: error.lineNumber,
+            };
+          } else {
+            console.error(error.stack || error);
+            body = {
+              type: 'InternalError',
+              description: 'react-packager has encountered an internal error, ' +
+                'please check your terminal error output for more details',
+            };
+          }
+
+          return forEachBundleEntry((bundleEntry, platform) => ({type: 'error', body}));
```

Comment:
Hmm not sure if it's correct to send the error to *all* the clients. What if you run into an adge case that affected only one of the clients?